### ISSUE TITLE
unnecessary_operation deletes labeled blocks

### ISSUE THREAD

**Issue by digama0** [rust-lang/rust-clippy#11575]
### Summary

The `unnecessary_operation` lint seems to delete labeled blocks, which causes a "break outside of loop" error. Curiously, in the example below, the lint only triggers with the `;` after the block; the statement version does not trigger the lint.

### Lint Name

unnecessary_operation

### Reproducer

I tried this code:

```rust
pub fn foo() -> u8 {
    'a: {
        break 'a
    };
    1
}
fn main() {}
```

I saw this happen:

```
warning: unnecessary operation
 --> src/main.rs:2:5
  |
2 | /     'a: {
3 | |         break 'a
4 | |     };
  | |______^ help: statement can be reduced to: `break 'a;`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_operation
  = note: `#[warn(clippy::unnecessary_operation)]` on by default

warning: `rust-test` (bin "rust-test") generated 1 warning (run `cargo clippy --fix --bin "rust-test"` to apply 1 suggestion)
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

I expected to see this happen: no warning

### Version

```text
rustc 1.74.0-nightly (69e97df5c 2023-08-26)
binary: rustc
commit-hash: 69e97df5ce571a777acd654ec3697ae8d25962ea
commit-date: 2023-08-26
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by abizjak** [comment]
I have hit a similar situation with the following code

```
fn a() {
    'skip: {
        if true {
            let true = true else {
                break 'skip;
            };
        }
    };
}
```
which triggers the following warning
```
 --> src/main.rs:2:5
  |
2 | /     'skip: {
3 | |         if true {
4 | |             let true = true else {
5 | |                 break 'skip;
6 | |             };
7 | |         }
8 | |     };
  | |______^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_operation
  = note: `#[warn(clippy::unnecessary_operation)]` on by default
help: statement can be reduced to
  |
2 ~     if true {
3 +             let true = true else {
4 +                 break 'skip;
5 +             };
6 +         };
```

However both this code
```rust
fn a() {
    'skip: {
            let true = true else {
                break 'skip;
            };
    };
}
```

and (notice the lack of semicolon)
```rust 
fn a() {
    'skip: {
        if true {
            let true = true else {
                break 'skip;
            };
        }
    }
}
```
do not trigger the warning.
